contracts/core/EntryPoint.sol

```diff
@@ -79,12 +80,33 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard,
     (uint256 collected) {
         uint256 preGas = gasleft();
         bytes memory context = getMemoryBytesFromOffset(opInfo.contextOffset);
-
-        try this.innerHandleOp(userOp.callData, opInfo, context) returns (
-            uint256 _actualGasCost
-        ) {
-            collected = _actualGasCost;
-        } catch {
+        uint saveFreePtr;
+        assembly {
+            saveFreePtr := mload(0x40)
+        }
+        bytes calldata callData = userOp.callData;
+        bytes memory innerCall;
+        bytes4 methodSig;
+        assembly {
+            let len := callData.length
+            if gt(len,3) {
+                methodSig := calldataload(callData.offset)
+            }
+        }
+        if (methodSig == IAccountExecute.executeUserOp.selector) {
+            bytes memory executeUserOp = abi.encodeCall(IAccountExecute.executeUserOp, (userOp, opInfo.userOpHash));
```

Comment:
I think this is actually a bit too awkward. When the wallet app wants to pass some params to the Smart Account (say it wants to know what is 2+2) so it will need to create a calldata that is `abi.encodeWithSelector("executeUserOp", "add", 2, 2)`.
However this `(string, uint,uint)` is not the correct method signature for the `executeUserOp` method.
I think that developers will (rightfully) feel like they need to `abi.encode` the correct method call, which is `executeUserOp(UserOperation,bytes32)`, and will be confused.

Reply:
the executeUserOp is not a real methodsig, but instead a MAGIC value at the beginning of the callData which is expected not to collide with any valid methodSig.
The executeUserOp method is expected to read more than just the calldata to execute (otherwise, it wouldn't use this mechanism, but normal `execute()`...

if all you want is an executeUserOp that performs just what "execute" does, then the encoding should probably be:

abi.encodePacked( executeUserOp.selector, abi.encodeCall( execute, (target,0,data)));

and executeUserOp would call the inner method as:
`address(this).call(userOp.callData[4:])`
but again, this is a very heavy way to achieve just what "execute" does.